cast/src/lib.rs

```diff
@@ -511,6 +511,41 @@ impl SimpleCast {
         })
     }
 
+    /// Converts wei into an eth amount
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::from_wei(1.into(), "gwei".to_string())?, "0.000000001");
+    ///     assert_eq!(Cast::from_wei(12340000005u64.into(), "gwei".to_string())?, "12.340000005");
+    ///     assert_eq!(Cast::from_wei(10.into(), "ether".to_string())?, "0.00000000000000001");
+    ///     assert_eq!(Cast::from_wei(100.into(), "eth".to_string())?, "0.0000000000000001");
+    ///     assert_eq!(Cast::from_wei(17.into(), "".to_string())?, "17");
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn from_wei(value: U256, unit: String) -> Result<String> {
```

Comment:
Have you considered using the `ethers_core::utils::parse_units` and `format_units` methods? https://github.com/gakonst/ethers-rs/blob/master/ethers-core/src/utils/mod.rs#L103-L147